Feature/#36 post page author

torahoo opened the pull request:
게시글에 글쓴이 정보가 보이게 끔 수정하였습니다.

torahoo committed:
feature/#36-PostPageAuthorId :
1. Post HTML 부분 수정

seungin2 committed:
#36 test :Test code 작성

seungin2 committed:
Merge remote-tracking branch 'origin/feature/#36-PostPageAuthorId' into feature/#36-PostPageAuthorId

torahoo committed:
feature/#36-PostPageAuthorId :
1. UserService에 findbyid 메서드 추가
2. Post Entity, PageDTO, AddDTO를 authorId 추가하여 수정
3. PostController에  authorId 수정한거에 맞게 수정
4. HTML에 user정보가 보이게끔 수정
6. testData init 수정

torahoo committed:
feature/#36-PostPageAuthorId :
1. static 폴더에 images 폴더추가 후 defaultProfile 사진 추가
2. PostPage, AddForm 수정

torahoo committed:
feature/#36-PostPageAuthorId :
1. test코드들 authorId 추가하여 수정

seungin2 reviewed (approved):
ok

Hoogokok commented on geulkkoli/src/test/java/com/geulkkoli/domain/post/ImplPostRepositoryTest.java:
interface PostRepository에 대한 테스트라고 볼 수 있으니 ImplPostRepositoryTest를 일반적인 PostRepositoryTest로 바꾸는 건 어떨 까요

Hoogokok commented on geulkkoli/src/test/java/com/geulkkoli/domain/post/PostServiceTest.java:
```diff
@@ -31,9 +47,16 @@ void updatePost() {
 //        log.info("onePost={}", onePost);
 //        Post update = new Post("title02", "body02", onePost.getNickName());
         log.info("@Test ::1 one={}", postService.findById(1L));
```
log를 선언한 이유가 뭔가요?

Hoogokok commented on geulkkoli/src/test/java/com/geulkkoli/domain/post/PostServiceTest.java:
```diff
@@ -1,28 +1,44 @@
+import java.util.List;
+
+/**
+ * TestDataInit Class 참조
```
TestDataInit 클래스의 목적은 화면의 보여줄 정보가 없어서 사용하는 용도였습니다. 결국 이 데이터는 DB에 데이터가 있어서 더 이상 불필요해 진다면 사라질  테스트 더미 데이터라고 볼 수 있습니다. 

이때 이 아래의 테스트들은 통과되지 않을 겁니다. 왜냐하면 이 테스트들에 필요한 값을 외부 클래스인 TestDataIntit에 의존하고 있기 때문 입니다. 

충분히 작성된 테스트는 보통 테스트 대상의 구현이 바뀌어도 테스트 코드 바뀌지 않는다고 합니다. 

하지만 아래의 테스트 코드들은 테스트 대상이 바뀐 것도 아닌데 실패가 예견되는 테스트 코드 입니다. 즉 충분히 작성된 게 아닌 테스트 코드인 셈 입니다.

https://mangkyu.tistory.com/144 이 블로그의 given. when then 패턴의 given의 주의해서 주입할 데이터를 골라보는 게 어떨까요?

takoeats reviewed (approved):
잘 했을거에요

godchaean reviewed (commented):
확인

torahoo committed:
feature/#36-PostPageAuthorId :
1. ImplPostRepositoryTest -> PostRepositoryTest로 변경
2. PostRepositoryTest를 TestDataInit과 상관없이 별개로 동작하게 끔 설정
-> @ActiveProfiles("test") 추가 후 TestDataInit 클래스 자체에 @Profiles("!test") 어노테이션 추가
3. PostRepositoryTest에 update 메서드 수정
